### PR TITLE
[FX-4717] Delete icon property

### PR DESCRIPTION
**Pull request by sashuk**
[FX-4717]

### Description

This pull request deletes long deprecated icon property from `List.Item` component

### How to test

- [Temploy](https://picasso.toptal.net/fx-4717-remove-custom-icons-on-list-item)
- All checks should pass

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Double check if `picasso-tailwind-merge` requires major update (check its `README.md`)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Self reviewed
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4717]: https://toptal-core.atlassian.net/browse/FX-4717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]

File: .changeset/quiet-coins-switch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso-list': major
```
Comment (AdrianContiu) [inline .changeset/quiet-coins-switch.md:2]: Shouldn't we add `'@toptal/picasso': patch` and the heading , as well?

Comment (TomasSlama) [reply .changeset/quiet-coins-switch.md:2]: This might be actually even major bump for Picasso 🤔

Comment (sashuk) [reply .changeset/quiet-coins-switch.md:2]: @TomasSlama Does it mean that every major bump in one of Picasso dependency should result in major bump of `@toptal/picasso`?

Comment (sashuk) [reply .changeset/quiet-coins-switch.md:2]: @AdrianContiu @TomasSlama Actually yes, as `@toptal/picasso` is "proxy" component, it still has the public API that is a combined from all APIs from exported component. The `'@toptal/picasso': major` was added, thanks!

Comment (TomasSlama) [reply .changeset/quiet-coins-switch.md:2]: @sashuk I also added detailed explanation on https://github.com/toptal/picasso/pull/4510#discussion_r1746867098

**Pull request merged**